server/util/util.ts

```diff
@@ -112,3 +112,65 @@ export const timestampColumnTransformer: ValueTransformer = {
 };
 
 export const oneDaySeconds = 24 * 60 * 60;
+
+export type TimeInterval =
+  | 'day'
+  | 'week'
+  | 'month'
+  | 'year';
+
+export function getEsTimeInterval(interval: TimeInterval) {
+  switch (interval) {
+    case 'day': return 'd';
+    case 'week': return 'w';
+    case 'month': return 'M';
+    case 'year': return 'y';
+    default:
+      throw new Error(`Unsupported time interval '${interval}'`);
```

Comment:
Is this possible since TypeScript should enforce one of the above cases?

Reply:
There are two eslint rules preventing the removal of this right now. One is just the `default-case` rule that always requires them - which might be ok to disable - but the other is the `consistent-return` rule, which I would be hesitant to disable or set to ignore undefined. As much as TS does try to enforce the value passed in, it's possible an erroneous value might get forced through, so it seems worth having these error checks in place just in case.